feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+        model.add_constr(xsum(is_row_covered[row] for row in data.rows) == num_row_covered)
+
+        # Constraint: If selected is given, discard columns that are not part of selection
+        for i in data.X:
+            if i not in selected:
+                model.add_constr(x[i] == 0)
+
+        # Constraint: limit number of selected to max_cover_size
+        model.add_constr(xsum(x[i] for i in data.X) <= self.selection_size)
+
+        # Objective: maximize "row" coverage (not the whole coverage of 1s)
+        model.objective = maximize(xsum(is_row_covered[row] for row in data.rows))
+
+        # Solve
+        model.verbose = 0
+        model.optimize()
+
+        # Solution
+        selected = [i for i in data.X if float(x[i].x) >= 0.99]
+
+        if self.verbose:
+            print("=" * 40)
+            print("MAX COVER OBJECTIVE:", model.objective_value)
+            print("NUM ROWS COVERED:", num_row_covered.x,
+                  "coverage: {:.2f}".format(num_row_covered.x / data.matrix.shape[0]))
+            print("SIZE:", len(selected),
+                  "reduction: {:.2f}".format((data.matrix.shape[1] - len(selected)) / data.matrix.shape[1]))
+            print("SELECTED:", selected)
+            print("STATUS:", model.status)
+            print("=" * 40)
+
+        assert model.status == OptimizationStatus.OPTIMAL
+
+        # Return
+        return selected
+
+    @staticmethod
+    def _validate_args(selection_size):
+        if selection_size is not None:
+            if selection_size <= 0:
+                raise ValueError("Selection size must be greater than zero.")
+
+
+# TODO not sure if this should use _BaseDispatcher or not. We can decide later
+class _TextBased(_BaseSupervisedSelector):
+    """
+    featrization and optimization are done in 'fit' function.
+    The 'transform' function is used to show the selected contents
+    """
+
+    def __init__(self, seed: int, num_features: Num,
+                 featurization_method: TextWiser, optimization_method: str, cost_metric: str,
+                 trials: int = 10):
+        # Call constructor of parent class _BaseSupervisedSelector
+        super().__init__(seed)
+
+        self.selected_features_ = None
+        self.num_features = num_features
+        self.featurization_method = featurization_method
+        self.selection_size = num_features
+        self.optimization_method = optimization_method
+        self.cost_metric = cost_metric
+        self.trials = trials
+        self.content_selector = ContentSelector(selection_size=num_features,
+                                                seed=self.seed, trials=trials, verbose=True)
+
+    def fit(self, data: pd.DataFrame, labels: pd.DataFrame) -> NoReturn:
+
+        # Call fit method from parent class _BaseSupervisedSelector
+        super().fit(data, labels)
+
+        # Get the text columns dynamically
+        text_columns = [col for col in data.columns if col.startswith("item")]
```

Comment:
what does this line do? And how can we select columns with a hardcoded column name? How can we guarantee columns will start with "item" string?

Reply:
Then on 615 we are turning columns again to subset with "selected" only, so cannot we loop df.columns on line 615 and remove this line?

Reply:
Removed line 608, and directly looped it through the data.columns